torch/_ops.py

```diff
@@ -99,39 +99,59 @@ def __getattr__(self, key):
         if key == '__file__':
             return 'torch.ops'
 
+        # ensure that query for dunder attributes that does not exist on
+        # opoverloadpacket but instead exists on the self._op object does not unnecessarily call
+        # `_get_operation_overload` (which is an expensive operation).
+        # This is done to prevent any potential slowdown. This list can be extended
+        # if there exists other attributes like `__name__` that only exist on self._op and not on the
+        # opoverloadpacket.
+        try:
+            if key.startswith('__') and key.endswith('__'):
+                return getattr(self._op, key)
+        except AttributeError:
+            # for consistency because it seems weird to
+            # throw an attribute error with a message containing
+            # an object name different from the one the attribute
+            # query was performed on.
+            raise AttributeError("'{}' object has no attribute '{}'".format(str(self), key)) from None
+
         try:
             use_key = '' if key == 'default' else key
             # TODO: disallow access to overloads registered by JIT
-            op_ = torch._C._get_operation_overload(self._qualified_op_name, use_key)
+            op_ = torch._C._get_operation_overload(
+                self._qualified_op_name, use_key)
             schema = torch._C._get_schema(self._qualified_op_name, use_key)
             overload = OpOverload(self, op_, schema)
             # cache the overload object
             setattr(self, key, overload)
             return overload
         except RuntimeError:
-            try:
-                # This is added to maintain bc in case the user queries an attribute that exists on `self._op`
-                # which used to be returned before instead of the OpOverloadPacket
-                out = getattr(self._op, key)
-                return out
-            except AttributeError:
-                raise AttributeError("'{}' object has no attribute '{}'".format(str(self), key)) from None
+            raise AttributeError(
+                "'{}' object has no attribute '{}'".format(str(self), key)
```

Comment:
Should we be clearer here that we failed to get an op overload for a given op and a given overload?